src/test/java/ttt/gui/TicTacToeRulesTest.java

```diff
@@ -78,10 +80,27 @@ public void initialisesGame() {
                 new CommandLinePlayerFactoryStub(players)
         );
         ticTacToeRules.initialiseGame(HUMAN_VS_HUMAN, "3");
+
         assertThat(ticTacToeRules.getBoard(), is(board));
     }
 
     @Test
+    public void currentPlayerReinitialisedWhenNewGameStarted(){
+        Board board = new Board(3);
+
+        TicTacToeRules ticTacToeRules = new TicTacToeRules(
+                new BoardFactoryStub(board, board),
+                new CommandLinePlayerFactoryStub(players)
+        );
+
+        ticTacToeRules.initialiseGame(HUMAN_VS_HUMAN, "3");
+        ticTacToeRules.takeTurn(1);
+        ticTacToeRules.initialiseGame(HUMAN_VS_UNBEATABLE, "3");
+
+        assertThat(ticTacToeRules.getCurrentPlayerIndex(), is(0));
+    }
```

Comment:
This looks like a lot of setup code to make sure that the players are initialised correctly. Could this be an indicator that there is some unnecessary complexity in the code? Or maybe we are using to many _"real"_ objects in this test?